perfmetrics/scripts/continuous_test/gcp_ubuntu/build.sh

```diff
@@ -39,17 +39,38 @@ then
   UPLOAD_FLAGS="--upload_gs"
 fi
 
+run_load_test_and_fetch_metrics(){
+  # Executing perf tests
+  ./run_load_test_and_fetch_metrics.sh "$1" "$2" "$3" "$4" "$5"
+
+  # ls_metrics test. This test does gcsfuse mount with the passed flags first and then does the testing.
+  LOG_FILE_LIST_TESTS=${KOKORO_ARTIFACTS_DIR}/gcsfuse-list-logs-$6.txt
+  GCSFUSE_LIST_FLAGS="$7 --log-file $LOG_FILE_LIST_TESTS"
+  cd "./ls_metrics"
+  ./run_ls_benchmark.sh "$GCSFUSE_LIST_FLAGS" "$2" "$4" "$8"
+  cd "../"
+}
+
 GCSFUSE_FLAGS="--implicit-dirs  --debug_fuse --debug_gcs --log-format \"text\" "
-LOG_FILE_FIO_TESTS=${KOKORO_ARTIFACTS_DIR}/gcsfuse-logs.txt
+LOG_FILE_FIO_TESTS=${KOKORO_ARTIFACTS_DIR}/gcsfuse-logs-flat.txt
 GCSFUSE_FIO_FLAGS="$GCSFUSE_FLAGS --log-file $LOG_FILE_FIO_TESTS --stackdriver-export-interval=30s"
 BUCKET_NAME="periodic-perf-tests"
 SPREADSHEET_ID='1kvHv1OBCzr9GnFxRu9RTJC7jjQjc9M4rAiDnhyak2Sg'
+GSHEET_CREDENTIALS="gs://periodic-perf-tests/creds.json"
+LIST_CONFIG_FILE="config.json"
+run_load_test_and_fetch_metrics "$GCSFUSE_FIO_FLAGS" "$UPLOAD_FLAGS" "$BUCKET_NAME" "$SPREADSHEET_ID" "$GSHEET_CREDENTIALS" "flat" "$GCSFUSE_FLAGS" "$LIST_CONFIG_FILE"
 
-# Executing perf tests
-./run_load_test_and_fetch_metrics.sh "$GCSFUSE_FIO_FLAGS" "$UPLOAD_FLAGS" "$BUCKET_NAME" "$SPREADSHEET_ID"
-
-# ls_metrics test. This test does gcsfuse mount with the passed flags first and then does the testing.
-LOG_FILE_LIST_TESTS=${KOKORO_ARTIFACTS_DIR}/gcsfuse-list-logs.txt
-GCSFUSE_LIST_FLAGS="$GCSFUSE_FLAGS --log-file $LOG_FILE_LIST_TESTS"
-cd "./ls_metrics"
-./run_ls_benchmark.sh "$GCSFUSE_LIST_FLAGS" "$UPLOAD_FLAGS" "$SPREADSHEET_ID"
+touch config.yml
+echo "enable-hns: true
+metadata-cache:
+  ttl-secs: 0" > config.yml
+COMMON_MOUNT_FLAGS="--debug_fuse --debug_gcs --log-format \"text\""
+GCSFUSE_LS_FLAGS="--config-file=../config.yml $COMMON_MOUNT_FLAGS"
+LOG_FILE_FIO_TESTS=${KOKORO_ARTIFACTS_DIR}/gcsfuse-logs-hns.txt
+GCSFUSE_FIO_FLAGS="--config-file=config.yml $COMMON_MOUNT_FLAGS --log-file $LOG_FILE_FIO_TESTS --stackdriver-export-interval=30s"
+BUCKET_NAME="periodic-perf-tests-hns"
+SPREADSHEET_ID='1wXRGYyAWvasU8U4KaP7NGPHEvgiOSgMd1sCLxsQUwf0'
```

Comment:
what is this hard coded id used for ?

Reply:
We have existing spreadsheet where the fio and ls metrics are getting uploaded  daily from the periodic tests, 
So for the flat bucket the metrics are getting uploaded(first hard coded id) [here](https://docs.google.com/spreadsheets/d/1kvHv1OBCzr9GnFxRu9RTJC7jjQjc9M4rAiDnhyak2Sg/edit?gid=0#gid=0).
For the hns bucket metrics i have created another copy of similar gsheet where the results will be uploaded. [Here](https://docs.google.com/spreadsheets/d/1wXRGYyAWvasU8U4KaP7NGPHEvgiOSgMd1sCLxsQUwf0/edit?resourcekey=0-Kv_559oGD9Z9474dPN6dxA&gid=0#gid=0). (currently this contains the results from the smaller workload that i used for testing the script)

Reply:
okay,so new runs will override the old result and add latest results in same spreadsheet every time...this spreadsheet link is provided in kokoro pipeline?

Reply:
Yes.